src/vm/zapsig.cpp

```diff
@@ -441,6 +441,12 @@ BOOL ZapSig::GetSignatureForTypeHandle(TypeHandle      handle,
         case ELEMENT_TYPE_CLASS:
         {            
             CorSigUncompressToken(pSig, &tk);
+            if (TypeFromToken(tk) == mdtTypeRef)
+            {
+                ENABLE_FORBID_GC_LOADER_USE_IN_THIS_SCOPE();
+                TypeHandle actualHandle = ClassLoader::LoadTypeDefOrRefThrowing(pModule, tk, ClassLoader::ReturnNullIfNotFound, ClassLoader::FailIfUninstDefOrRef, tdAllTypes);
```

Comment:
This should use ResolveTokenToTypeDefThrowing to avoid doing full load of a wrong type. And we may do try/catch around it and swallow any exceptions - in case there is a problem resolving token for a wrong type.